domain/test/usecases/authentication/upload_file_interactor_test.dart

```diff
@@ -0,0 +1,119 @@
+import '../../mock/repository/authentication/mock_document_repository.dart';
+import '../../mock/repository/authentication/mock_token_repository.dart';
+
+void main() {
+  group('upload_file_interactor_test', () {
+    MockDocumentRepository documentRepository;
+    MockTokenRepository tokenRepository;
+    MockCredentialRepository credentialRepository;
+    UploadFileInteractor uploadFileInteractor;
+    UploadTaskId uploadTaskId;
+
+    setUp(() {
+      documentRepository = MockDocumentRepository();
+      tokenRepository = MockTokenRepository();
+      credentialRepository = MockCredentialRepository();
+      uploadTaskId = UploadTaskId('upload_task_id_1');
+      uploadFileInteractor = UploadFileInteractor(documentRepository, tokenRepository, credentialRepository);
+    });
+
+    test('uploadFileInteractorShouldReturnSuccessWithCorrectData', () async {
+      when(tokenRepository.getToken()).thenAnswer((_) async => permanentToken);
+      when(credentialRepository.getBaseUrl()).thenAnswer((_) async => linShareBaseUrl);
+      when(documentRepository.upload(fileInfo1, permanentToken, linShareBaseUrl))
+          .thenAnswer((_) async => FileUploadState(
+              Stream.fromIterable([
+                Right<Failure, Success>(fileUploadProgress10),
+                Right<Failure, Success>(fileUploadProgress100),
+                Right<Failure, Success>(FileUploadSuccess())
+              ]),
+              uploadTaskId));
+
+      final resultStream = uploadFileInteractor.execute(fileInfo1);
+
+      expect(
+          resultStream,
+          emitsInOrder([
+            Right<Failure, Success>(PreparingUpload(fileInfo1)),
+            Right<Failure, Success>(UploadingProgress(fileUploadProgress10.progress, fileInfo1.fileName)),
+            Right<Failure, Success>(UploadingProgress(fileUploadProgress100.progress, fileInfo1.fileName)),
+            Right<Failure, Success>(UploadFileSuccess(fileInfo1))
+          ]));
+    });
+
+    test('uploadFileInteractorShouldFailureWithWrongBaseUrl', () async {
+      when(tokenRepository.getToken()).thenAnswer((_) async => permanentToken);
+      when(credentialRepository.getBaseUrl()).thenAnswer((_) async => wrongUrl);
+      final exception = Exception();
+      when(documentRepository.upload(fileInfo1, permanentToken, wrongUrl))
+          .thenThrow(exception);
+
+      final resultStream = uploadFileInteractor.execute(fileInfo1);
+
+      expect(
+          resultStream,
+          emitsInOrder([
+            Right<Failure, Success>(PreparingUpload(fileInfo1)),
+            Left<Failure, Success>(UploadFileFailure(exception))
+          ]));
+    });
+
+    test('uploadFileInteractorShouldFailureWithWrongToken', () async {
+      final wrongToken = Token('token', TokenId('uuid'));
+      when(tokenRepository.getToken()).thenAnswer((_) async => wrongToken);
+      when(credentialRepository.getBaseUrl()).thenAnswer((_) async => linShareBaseUrl);
```

Comment:
How about if `ServerNotFound`, `TimeoutConnect`, ...or any error with server?

Reply:
Ok I will add this later when I update `datasource`